pytensor/link/pytorch/dispatch/shape.py

```diff
@@ -0,0 +1,54 @@
+import torch
+
+from pytensor.link.pytorch.dispatch.basic import pytorch_funcify
+from pytensor.tensor.shape import Reshape, Shape, Shape_i, SpecifyShape, Unbroadcast
+
+
+@pytorch_funcify.register(Reshape)
+def pytorch_funcify_Reshape(op, node, **kwargs):
+    shape = node.inputs[1]
+
+    def reshape(x, shape=shape):
+        return torch.reshape(x, tuple(shape))
+
+    return reshape
```

Comment:
We have to use the runtime shape, since it is not always a constant.

```suggestion
@pytorch_funcify.register(Reshape)
def pytorch_funcify_Reshape(op, node, **kwargs):
      def reshape(x, shape):
        return torch.reshape(x, tuple(shape))

    return reshape
```